frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt

```diff
@@ -154,6 +158,21 @@ class MainActivity : Activity() {
           queryInput.typeface = if (empty) Typeface.DEFAULT else robotoMono
         }
 
+    var totalDy = 0
+    val vc = ViewConfiguration.get(recycler.context)
+    val slop = vc.scaledTouchSlop
+    recycler.scrollEvents()
+            .filter({ scroll -> scroll.dy() > 0 })
+            .observeOn(mainThread())
+            .crashingSubscribe {
+              scroll -> totalDy += scroll.dy()
+              if (totalDy >= slop) {
+                val inputManager = applicationContext.getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
```

Comment:
How about using `systemService` extension function found in utils?
```
    systemService<InputMethodManager>().apply {
        hideSoftInputFromWindow(currentFocus?.windowToken, HIDE_NOT_ALWAYS)
    }
```

Reply:
Cool! Didn't know about this. I'll update this.